src/vs/workbench/contrib/terminal/browser/terminalService.ts

```diff
@@ -347,42 +384,21 @@ export class TerminalService implements ITerminalService {
 	}
 
 	public async initializeTerminals(): Promise<void> {
-		const enableTerminalReconnection = this.configHelper.config.enablePersistentSessions;
-		const serverSpawn = this.configHelper.config.serverSpawn;
-		if (!!this._environmentService.remoteAuthority && enableTerminalReconnection && serverSpawn) {
-			let emptyTab: TerminalTab | undefined;
+		if (this._remoteTerminalsInitialized) {
 			if (!this.terminalTabs.length) {
-				emptyTab = this._instantiationService.createInstance(TerminalTab, this._terminalContainer, undefined);
-				this._terminalTabs.push(emptyTab);
+				this._tempEmptyTab = this._instantiationService.createInstance(TerminalTab, this._terminalContainer, undefined);
+				this._terminalTabs.push(this._tempEmptyTab);
 				this._onInstanceTitleChanged.fire(undefined);
 			}
```

Comment:
So it seems like the empty tab is only needed because the terminal lacks proper UI for a "connecting" state as before this work the "starting" state would only apply when valid terminal creation requests were active (mainly the restore panel initial terminal).

How about making the dropdown aware of this initializing state and adding in the fake terminal instead since this already depends on the terminal service here:

https://github.com/microsoft/vscode/blob/7210c7246ab3dd56492d5956ca4158a408e1f0b0/src/vs/workbench/contrib/terminal/browser/terminalActions.ts#L298-L384

Note that the way terminals in the dropdown are activated is based on regex matching on the title name here, just be careful of that:

https://github.com/microsoft/vscode/blob/7210c7246ab3dd56492d5956ca4158a408e1f0b0/src/vs/workbench/contrib/terminal/browser/terminalActions.ts#L325-L329

Reply:
I can try that. It's also for the focus thing which is hard. Maybe the TerminalService can hold that state.

Reply:
Oh to catch the case where users focus the terminal quickly and starts typing? If it breaks that, maybe we should just polish up this empty tab idea then. My main opposition to it is the fact that we have the `_tempEmptyTab` property in terminal service when it's only really useful during start up. How about instead we check to see if the first tab has `TerminalTab.terminalInstances === 0` and treat that as the empty tab, since I think that's the only way to get into that state.

Reply:
> Oh to catch the case where users focus the terminal quickly and starts typing?

This won't capture the actual input, you would need a full TerminalInstance for that, it only knows to focus the terminal instance once it exists

Reply:
Oh in that case, having a look.